create a single block mapping filter and a single thread mapping filter per branch

skimo-openhub opened the pull request:
After these changes, there is only a single call left to `makeMappingFilter`
(in `MappedScop::map`).  This will make it easy to change the representation
of the mapping filter to contain extra information available inside `MappedScop::map`.
The change of the representation itself is left for a later PR.

ftynse commented on tc/core/polyhedral/cuda/mapped_scop.cc:
```diff
@@ -107,42 +107,47 @@ const CudaDim& mappingSize<mapping::ThreadId>(const MappedScop* mscop) {
+    const std::vector<MappingIdType>& ids,
+    const CudaDimView& extent) {
+  CHECK_EQ(ids.size(), list.n());
+  CHECK_LE(list.n(), extent.size()) << "dimension overflow";
```
I would add a function-level comment mentioning this assumption: if "extent" has more values than "ids", only the first ids.size() values of "extent" will be used.

ftynse commented on tc/core/polyhedral/cuda/mapped_scop.cc:
```diff
@@ -107,42 +107,47 @@ const CudaDim& mappingSize<mapping::ThreadId>(const MappedScop* mscop) {
-  auto mapping =
-      detail::ScheduleTree::makeMappingFilter<MappingIdType>(filter, {id});
+    // Create mapping filter by equating the newly introduced
+    // parameter "id" to the "pos"-th schedule dimension modulo its extent.
```
this comment is outdated

ftynse commented on tc/core/polyhedral/cuda/mapped_scop.h:
```diff
@@ -134,6 +134,14 @@ class MappedScop {
 
  private:
+  // Map the elements in "list" to successive blocks or thread identifiers,
+  // with the first element mapped identifier X.
```
mapped _to_ identifier X

skimo-openhub commented:
On Fri, Apr 27, 2018 at 08:53:50AM +0000, ftynse wrote:
> ftynse commented on this pull request.
> 
> shouldn't 384ccc4f9d170224c3b7f049f8ef9825db7b0428 (Bump isl for isl::union_pw_aff_list::reverse and isl:union_pw_aff_involves_param) go before b937b5b211849c1b4b8bd42b0188761418327246 that uses these functions?

It does come before (in history, at least; perhaps you are confused
by the fact that github ignores history)

skimo

ftynse commented:
> It does come before (in history, at least; perhaps you are confused by the fact that github ignores history)

Yes, I'm looking at commits in the order of github... Nevermind then.

skimo-openhub commented:
f84119cc74ab8ec09bab2186d491faf763b57326: fixed typo in commit message
51229f5c9996c5ba57cbe828f306beaa269c9fc2: updated comments
bff75f5e5ef71acf850a7715f07dc0b92a78a7c5: updated comments

diff:
```
diff --git a/tc/core/polyhedral/cuda/mapped_scop.cc b/tc/core/polyhedral/cuda/mapped_scop.cc
index 627299f5..a44abffb 100644
--- a/tc/core/polyhedral/cuda/mapped_scop.cc
+++ b/tc/core/polyhedral/cuda/mapped_scop.cc
@@ -109,8 +109,8 @@ const CudaDim& mappingSize<mapping::ThreadId>(const MappedScop* mscop) {
 } // namespace
 
 // Map the elements in "list" to successive blocks or thread identifiers,
-// with the first element mapped to identifier X.
-// The extents are obtained from numBlocks or numThreads.  The identifiers
+// with the first element mapped to identifier X.  The extents are obtained
+// from the initial elements of numBlocks or numThreads.  The identifiers
 // must not be present in the space of the partial schedules in "list" and
 // extents must be non-zero.  The mapping corresponds to inserting a filter
 // node with condition 'list % extent = ids'.
@@ -140,7 +140,7 @@ detail::ScheduleTree* MappedScop::map(
     CHECK_NE(extent[i], 0u) << "NYI: mapping to 0";
 
     // Create mapping filter by equating the newly introduced
-    // parameter "id" to the "pos"-th schedule dimension modulo its extent.
+    // parameter ids[i] to the "i"-th affine function modulo its extent.
     upa = upa.mod_val(isl::val(tree->ctx_, extent[i]));
     upa = upa.sub(isl::union_pw_aff::param_on_domain(domain, id));
     filter = filter.intersect(upa.zero_union_set());
diff --git a/tc/core/polyhedral/cuda/mapped_scop.h b/tc/core/polyhedral/cuda/mapped_scop.h
index e9c08581..0ebd5d4c 100644
--- a/tc/core/polyhedral/cuda/mapped_scop.h
+++ b/tc/core/polyhedral/cuda/mapped_scop.h
@@ -129,7 +129,7 @@ class MappedScop {
 
  private:
   // Map the elements in "list" to successive blocks or thread identifiers,
-  // with the first element mapped identifier X.
+  // with the first element mapped to identifier X.
   // Return a pointer to the updated node (below the inserted filter)
   // for call chaining purposes.
   template <typename MappingTypeId>
```

ftynse commented:
Running tests....

ftynse commented:
tests pass locally /@ftynse
